scala/private/rule_impls.bzl

```diff
@@ -394,11 +395,27 @@ def _compile_or_empty(
         jars2labels,
         implicit_junit_deps_needed_for_java_compilation,
         unused_dependency_checker_mode,
-        unused_dependency_checker_ignored_targets):
+        unused_dependency_checker_ignored_targets,
+        deps_providers = []):
     # We assume that if a srcjar is present, it is not empty
     if len(ctx.files.srcs) + len(srcjars.to_list()) == 0:
         _build_nosrc_jar(ctx)
 
+        exports = []
```

Comment:
should this be in a method? It's almost the same code below (apart from the ijar thing which actually confused me)

Reply:
Yup, done.